plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/ptf/RawQuery.java

```diff
@@ -0,0 +1,174 @@
+                                    .build(),
+                            ScalarArgumentSpecification.builder()
+                                    .name("COLLECTION")
+                                    .type(VARCHAR)
+                                    .build(),
+                            ScalarArgumentSpecification.builder()
+                                    .name("FILTER")
+                                    .type(VARCHAR)
+                                    .build()),
+                    GENERIC_TABLE);
+            this.metadata = requireNonNull(metadata, "metadata is null");
+        }
+
+        @Override
+        public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+        {
+            String database = ((Slice) ((ScalarArgument) arguments.get("DATABASE")).getValue()).toStringUtf8();
+            String collection = ((Slice) ((ScalarArgument) arguments.get("COLLECTION")).getValue()).toStringUtf8();
+            String filter = ((Slice) ((ScalarArgument) arguments.get("FILTER")).getValue()).toStringUtf8();
+            // TODO: Improve case sensitivity handling
+            if (!database.equals(database.toLowerCase(ENGLISH))) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Only lowercase database name is supported");
+            }
+            if (!collection.equals(collection.toLowerCase(ENGLISH))) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Only lowercase collection name is supported");
+            }
+            if (!isValidJson(filter)) {
+                throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Can't parse 'filter' argument as json");
+            }
+
+            MongoTableHandle tableHandle = new MongoTableHandle(QUERY, new SchemaTableName(database, collection), Optional.of(filter));
+            ConnectorTableSchema tableSchema = metadata.getTableSchema(session, tableHandle);
+            Map<String, ColumnHandle> columnsByName = metadata.getColumnHandles(session, tableHandle);
+            List<ColumnHandle> columns = tableSchema.getColumns().stream()
+                    .map(ColumnSchema::getName)
+                    .map(columnsByName::get)
+                    .collect(toImmutableList());
+
+            Descriptor returnedType = new Descriptor(columns.stream()
+                    .map(MongoColumnHandle.class::cast)
+                    .map(column -> new Descriptor.Field(column.getName(), Optional.of(column.getType())))
+                    .collect(toList()));
+
+            RawQueryFunctionHandle handle = new RawQueryFunctionHandle(tableHandle);
+
+            return TableFunctionAnalysis.builder()
+                    .returnedType(returnedType)
+                    .handle(handle)
+                    .build();
+        }
+    }
+
+    private static boolean isValidJson(String json)
+    {
+        try {
+            Document.parse(json);
```

Comment:
is this `Document` is serializable/de-serializable via Jackson, if so we can maintain it as `Document` in `MongoTableHandle`